client/application/home/home.controller.js

```diff
@@ -4,136 +4,193 @@
+      vm.makeReservation = makeReservation;
+      vm.modifyReservation = modifyReservation;
+      vm.deleteReservation = deleteReservation;
+      vm.resetCache = resetCache;
+      vm.setCache = setCache;
+      vm.dateToNumber = HomeService.dateToNumber;
       vm.calendarView = 'week';
       vm.viewDate = new Date();
       vm.toggleText = 'Show Room List';
+      vm.message = 'Select a timeslot to start reservation';
       vm.roomList = [];
       vm.roomNumber = '';
       vm.cellIsOpen = true;
       vm.events = [];
       vm.myReservations = [];
+      vm.myWaitingList = [];
       initData();
+      $interval(getRoomInfo, 1500);
     }
 
-    // check login and init all data for the view
-
     function initData() {
+      vm.resetCache();
       ApiService.account('checkLogin').then(function(loggedInUser) {
+        currentUser = loggedInUser.success.username;
         vm.authenticated = true;
         ApiService.booking('getRoomList').then(function(roomList) {
           vm.roomList = roomList.rooms;
           vm.roomNumber = vm.roomList[0];
           getRoomInfo();
-        });
-        ApiService.booking('getMyReservation', {
-          userId: loggedInUser.success.userId
-        }).then(function(myReservations) {
-            vm.myReservations = myReservations.reservations;
+          getMyInfo();
         });
       }, function() {
         $state.go('login');
       });
     }
 
-    // parse return data into calendar event object
-
-    function createEvent(reservation, type) {
-      var dateString = reservation.timeslot.date.replace('GMT', 'EST');
-      var start = new Date(dateString);
-      start.setHours(reservation.timeslot.startTime);
-      var end = new Date(dateString);
-      end.setHours(reservation.timeslot.endTime);
-      var event = {};
-      event.title = reservation.description;
-      event.startsAt = start;
-      event.endsAt = end;
-      event.color = type;
-      return event;
-    }
-
-    // get reservation, waitingList, and equipment of a room
-
     function getRoomInfo() {
-      vm.events = [];
+      var tempEvents = [];
       ApiService.booking('getAllReservation', {
         roomId: vm.roomNumber
       }).then(function(res){
-        var reservations = res.reservations;
+        var reservations = res.reservations || [];
+        var waitingList = res.waitingList || [];
         for (var i = 0; i < reservations.length; i++) {
-          vm.events.push(createEvent(reservations[i], calendarConfig.colorTypes.info));
+          tempEvents.push(HomeService.createEvent(reservations[i], calendarConfig.colorTypes.info));
         }
+        for (var j = 0; j < waitingList.length; j++) {
+          tempEvents.push(HomeService.createEvent(reservations[i], calendarConfig.colorTypes.warning));
+        }
+        vm.events = tempEvents;
       });
-      // TODO: get equipment list
     }
 
-    /*
-    UI Actions
-    */
+    function getMyInfo() {
+      ApiService.booking('getMyReservation', {
```

Comment:
I use single for variables and function names, so I will keep them in consistency